bits/reboot_hook.s

```diff
@@ -83,6 +84,9 @@ koffset_base:    .word kernel_code-jump_to_kernel-12
 .pool
 
 kernel_code:
+
+    ldr sp, =0x080FF000
+
```

Comment:
This line shouldn't be needed.

Immediately after this, we jump to the payload, and sp is overwritten before any stack operations are performed.